pallets/gear/src/benchmarking/mod.rs

```diff
@@ -278,7 +296,84 @@ where
         pages
     };
 
-    (session_id, memory_pages)
+    (get_last_session_id::<T>().unwrap(), memory_pages)
+}
+
+#[track_caller]
+fn tasks_send_user_message_prepare<T: Config>(delay: u32)
+where
+    T::AccountId: Origin,
+{
+    use demo_delayed_sender::WASM_BINARY;
```

Comment:
Just wanna to know)) whether it's so difficult to make wat (actually is a vector of instruction) example for tasks, like in all other benchmarks. Instead of using demos, which where made for other goals. I mean that demos can have some side effects, which can affect on benchmark and without investigation it's hard to understand that all right.

Reply:
Totally agree. But we can avoid wats at all.
1. Put program in storage
2. Call ext_manager's required JournalNote (that you build manually just in place) -> it leads to some valid storage condition
3. Call manually processing of DelayedTask

Reply:
Nope, cannot agree with you. Examples were introduces to ease business logic in various testing component (and in benchmarks  too) so I use them this very way.

About side effects: samples are used to create all required conditions for task however we bench only task itself without any additional logic. It means there is no possible logic that could affect task execution.